### PR TITLE
Simple version of Explicit Boundary Histogram.

### PR DESCRIPTION
**Pull request by victlu**
Add a simple implementation of an Explicit Boundary Histogram metric aggregator.

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            {
+                int len = 0;
+                foreach (var bucket in agg.Buckets)
+                {
+                    Assert.Equal(1, bucket.Count);
+                    len++;
+                }
+                Assert.Equal(2, len);
+            }
+        }
```
Comment (pirgeo) [inline test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs:85]:
```suggestion
        }
        
        [Fact]
        public void HistogramWithEmptyBuckets()
        {
            using var meter = new Meter("TestMeter", "0.0.1");

            var hist = new HistogramMetricAggregator("test", "desc", "1", meter, DateTimeOffset.UtcNow, new KeyValuePair<string, object>[0], new double[] { 0, 5, 10 });

            hist.Update<long>(-3);
            hist.Update<long>(-2);
            hist.Update<long>(-1);
            hist.Update<long>(6);
            hist.Update<long>(7);
            hist.Update<long>(12);
            var metric = hist.Collect(DateTimeOffset.UtcNow, false);

            Assert.NotNull(metric);
            Assert.IsType<HistogramMetricAggregator>(metric);

            if (metric is HistogramMetricAggregator agg)
            {
                var expectedCounts = new int[] { 3, 0, 2, 0, 1 };
                int len = 0;
                foreach (var bucket in agg.Buckets)
                {
                    if (len < expectedCounts.Length) {
                        Assert.Equal(expectedCounts[len], bucket.Count);
                        len++;
                    }
                }
                Assert.Equal(5, len);
            }
```

How about adding something like this?

**Review by pirgeo — APPROVED** [review]

**Comment by cijothomas** [comment]
@victlu could you fix the CI failures, when you get a chance, and we are good to merge.

File: test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs (hunk 1 above)
Comment (victlu) [reply test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs:85]: Looks good! I'll add it!

**Comment by victlu** [comment]
Technical issues. Created new PR in:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2216

**Pull request closed without merging**